cvpy/annotation/base/Task.py

```diff
@@ -1,9 +1,20 @@
+from cvpy.base.ImageTable import ImageTable
+from cvpy.annotation.base.Project import Project
+
+
 class Task(object):
 
-    def __init__(self, task_id: str, start_image_id: str, end_image_id: str) -> None:
-        self._task_id = task_id
-        self._start_image_id = start_image_id
-        self._end_image_id = end_image_id
+    def __init__(self, start_image_id: int = None, end_image_id: int = None,
+                 image_table: ImageTable = None, project: Project = None) -> None:
+        self._task_id = None
+        self._project = project
+        self._image_table = image_table
+        if image_table:
+            self._start_image_id = 0
+            self._end_image_id = image_table.table.tableinfo().TableInfo.Rows.values[0] - 1
+        else:
+            self._start_image_id = start_image_id
+            self._end_image_id = end_image_id
```

Comment:
Instead of checking for image_table, the above logic should check if start_image_id/end_image_id are None. Otherwise, a user cannot set start_image_id/end_image_id along with image_table, which is how a user would typically create a Task object. 

```
if start_image_id is None:
  # set start_image_id = 0

if end_image_id is None:
  # set end_image_id = (image count - 1)
```

Also, if start_image_id/end_image_id are not None, then it will be good to check if there are as many IDs as the number of images in the image_table.

Reply:
I agree, checking that the number of ID's matches the number of images is a good idea. Question, just how flexible do you want this? Should a user be able to specify just the start ID? Just the end ID? In either of those situations should I adjust the other ID to be correct, e.g. with 5 images and a user supplied end ID of 10, should I set the start ID to 6? If we want this I will need to add additional checks to ensure we generate valid ID's.

Reply:
Per our discussion, the start/end ID's are now [0, num_rows_in_cas_table - 1].